### ISSUE TITLE
`no-unused-vars` vs `arguments` vs default parameters

### ISSUE THREAD

**Issue by bfred-it** [eslint/eslint#6726]
**What version of ESLint are you using?**

3.1.1

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```
eslint
--no-eslintrc
--env es6
--rule 'no-unused-vars: 2'
input.js
```

**What did you do? Please include the actual source code causing the issue.**

```js
function test(foo, bar = "hello") {
	console.log(arguments);
}
test();
```

**What did you expect to happen?**

No error, `bar` is _used_ by `arguments` and it has to be defined because I'm setting its default value.

**What actually happened? Please include the actual, raw output from ESLint.**

```log
/input.js
  1:20  error  'bar' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mysticatea** [comment]
Thank you for this issue.

As far as I know, default parameters don't affect `arguments` variable. You can remove `bar` safely.

    $ node -e "function foo(a = 1) { console.log(arguments); } foo()"
    {}

**Comment by bfred-it** [comment]
Ouch, I didn't even test the function. That's unexpected but it's ES6' fault :(
